govcd/disk.go

```diff
@@ -429,3 +437,50 @@ func (vdc *Vdc) GetDiskById(diskId string, refresh bool) (*Disk, error) {
 	}
 	return nil, ErrorEntityNotFound
 }
+
+// Get a VMs HREFs that is attached to the disk
+// An independent disk can be attached to at most one virtual machine.
+// If the disk isn't attached to any VM, return empty slice.
+// Otherwise return the list of VMs HREFs.
+func (disk *Disk) GetAttachedVmsHrefs() ([]string, error) {
+	util.Logger.Printf("[TRACE] GetAttachedVmsHrefs, HREF: %s\n", disk.Disk.HREF)
+
+	var vmHrefs []string
+
+	var attachedVMsLink *types.Link
+
+	// Find the proper link for request
+	for _, diskLink := range disk.Disk.Link {
+		if diskLink.Type == types.MimeVMs {
+			util.Logger.Printf("[TRACE] GetAttachedVmsHrefs - found the proper link for request, HREF: %s, name: %s, type: %s,id: %s, rel: %s \n",
+				diskLink.HREF, diskLink.Name, diskLink.Type, diskLink.ID, diskLink.Rel)
+
+			attachedVMsLink = diskLink
+			break
+		}
+	}
+
+	if attachedVMsLink == nil {
+		return nil, fmt.Errorf("[ERROR] GetAttachedVmsHrefs - could not find request URL for attached vm in disk Link")
```

Comment:
We usually used word `error` for errors. Not sure if this is just because it got confused with logging message or a specific reason.

Reply:
updated